res/skins/QMLDemo/Sampler.qml

```diff
@@ -0,0 +1,176 @@
+
+        anchors.margins: 5
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.right: vuMeter.left
+        anchors.bottom: parent.bottom
+    }
+
+    Skin.ControlMiniKnob {
+        id: gainKnob
+
+        anchors.margins: 5
+        anchors.top: parent.top
+        anchors.right: parent.right
+        height: 40
+        width: 40
+        group: root.group
+        key: "pregain"
+        color: Theme.samplerColor
+    }
+
+    Skin.ControlButton {
+        id: playButton
+
+        anchors.top: embedded.top
+        anchors.left: embedded.left
+        activeColor: Theme.samplerColor
+        width: 40
+        height: 40
+        text: "Play"
+        group: root.group
+        key: "cue_gotoandplay"
+        highlight: playControl.playing
+    }
+
+    Text {
+        id: label
+
+        text: root.deckPlayer.title
+        anchors.top: embedded.top
+        anchors.left: playButton.right
+        anchors.right: embedded.right
+        anchors.margins: 5
+        elide: Text.ElideRight
+        font.family: Theme.fontFamily
+        font.pixelSize: Theme.textFontPixelSize
+        color: Theme.deckTextColor
+    }
+
+    Rectangle {
+        id: progressContainer
+
+        anchors.margins: 5
+        anchors.left: playButton.right
+        anchors.right: embedded.right
+        anchors.bottom: embedded.bottom
+        height: 5
+        radius: height / 2
+        color: "transparent"
+        border.color: Theme.deckLineColor
+        border.width: 1
+
+        Rectangle {
+            antialiasing: false // for performance reasons
+            width: playPositionControl.value * parent.width
+            height: parent.height
+            radius: height / 2
+            color: Theme.samplerColor
+
+            Mixxx.ControlProxy {
+                id: playPositionControl
+
+                group: root.group
+                key: "playposition"
+            }
+
+        }
+
+        MouseArea {
+            anchors.fill: progressContainer
+            cursorShape: Qt.PointingHandCursor
+            hoverEnabled: true
+            onPressed: {
+                playPositionControl.value = mouse.x / width;
+            }
+            onPositionChanged: {
+                if (containsPress)
+                    playPositionControl.value = mouse.x / width;
+
+            }
+        }
+
+    }
+
+    Skin.VuMeter {
+        id: vuMeter
+
+        group: root.group
+        key: "VuMeter"
+        width: 4
+        anchors.margins: 5
+        anchors.top: parent.top
+        anchors.bottom: parent.bottom
+        anchors.right: gainKnob.left
+    }
+
+    Mixxx.ControlProxy {
+        id: playControl
+
+        readonly property bool playing: value != 0
+
+        function stop() {
+            value = 0;
+        }
+
+        group: root.group
+        key: "play"
+    }
+
+    Mixxx.ControlProxy {
+        id: ejectControl
+
+        function trigger() {
+            value = 1;
+            value = 0;
```

Comment:
I'm not sure, but I think setting the value twice is unnecessary. Whats important is just that the signal for the change in value gets emitted. Even if the value wasn't actually changed.

Reply:
But it kind of seems hacky to me tbh. Maybe the cpp ControlProxy object should have a trigger method to emit a value change signal without changing anything.

Reply:
Yes. The CO interface is a bit strange for that though. IIRC you can configure a CO to swallow such signals when the value doesn't change as well. I guess this was a design decision made driven by the old Skin system...